src/libraries/System.Memory/tests/ReadOnlySpan/Count.byte.cs

```diff
@@ -0,0 +1,313 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        [Fact]
+        public static void ZeroLengthCount_Byte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(0);
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void ZeroLengthCount_RosByte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(new ReadOnlySpan<byte>(new byte[] { 0, 0 }));
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_Byte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int i = 0; i < length; i++)
```

Comment:
Why is this inner loop needed?

Reply:
I don't know, copied from DefaultFilledContains_Byte. Shall I remove them both?

Reply:
Yup, that loop is also unnecessary.